plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/PartitionData.java

```diff
@@ -139,8 +139,14 @@ public static Object getValue(JsonNode partitionValue, Type type)
             case TIME:
                 return partitionValue.asLong();
             case FLOAT:
+                if (partitionValue.asText().equals("NaN")) {
```

Comment:
can the nan be written as "nan" or "naN" ?

Reply:
It won't be as far as I read Iceberg and Avro codes.